src/pybind/mgr/dashboard/controllers/prometheus.py

```diff
@@ -75,11 +83,25 @@ def get_access_info(self, module_name):
                     user = access_info['user']
                     password = access_info['password']
                     certificate = access_info['certificate']
-                    cert_file = tempfile.NamedTemporaryFile(delete=False)
-                    cert_file.write(certificate.encode('utf-8'))
-                    cert_file.flush()
-
-        return user, password, cert_file
+                    ca_cert_file = tempfile.NamedTemporaryFile(delete=False)
+                    ca_cert_file.write(certificate.encode('utf-8'))
+                    ca_cert_file.flush()
+
+                    cert_file = None
+                    cert = mgr.get_localized_store("crt")  # type: ignore
+                    if cert is not None:
+                        cert_file = tempfile.NamedTemporaryFile(delete=False)
+                        cert_file.write(cert.encode('utf-8'))
+                        cert_file.flush()  # cert_tmp must not be gc'ed
```

Comment:
probably this could be also in its own function as its used repeatedely the same way in the code.

Reply:
fixed